### PR TITLE
relocating mixControls_ form FlowGenericProblem to FlowProblem

### PR DESCRIPTION
**Pull request by GitPaean**
so FlowGenericProblem can be used for non-blackoil setting. mixControls_ for now is designed for blackoil setting.

### CODE REVIEW

File: opm/simulators/flow/FlowGenericProblem_impl.hpp [hunk 1]
```diff
@@ -762,8 +750,7 @@ operator==(const FlowGenericProblem& rhs) const
            this->solventSaturation_ == rhs.solventSaturation_ &&
            this->solventRsw_ == rhs.solventRsw_ &&
            this->polymer_ == rhs.polymer_ &&
-           this->micp_ == rhs.micp_ &&
-           this->mixControls_ == rhs.mixControls_;
+           this->micp_ == rhs.micp_;
```
Comment (GitPaean) [inline opm/simulators/flow/FlowGenericProblem_impl.hpp:753]: I did not figure out whether FlowProblem needs a `operator==`, at least, it is not there yet.

File: opm/simulators/flow/FlowGenericProblem_impl.hpp [hunk 2]
```diff
@@ -107,7 +106,6 @@ serializationTestObject(const EclipseState& eclState,
     result.solventRsw_ = {18.0};
     result.polymer_ = PolymerSolutionContainer<Scalar>::serializationTestObject();
     result.micp_ = MICPSolutionContainer<Scalar>::serializationTestObject();
-    result.mixControls_ = MixingRateControls<FluidSystem>::serializationTestObject(schedule);
```
Comment (GitPaean) [inline opm/simulators/flow/FlowGenericProblem_impl.hpp:110]: Do we need this also for FlowProblem?

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/flow/FlowGenericProblem_impl.hpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/flow/FlowGenericProblem_impl.hpp:110]: There is no test for a full blown FlowProblem, only for the GenericFlowProblem (see test_RestartSerialization.cpp). Hence, there is no place to stick this as such.

I'd suggest adding a test for the MixingRateControls and leave it at that. Adding a test for the full problem class is.. painful.

File: opm/simulators/flow/FlowGenericProblem_impl.hpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/flow/FlowGenericProblem_impl.hpp:753]: Again as there is no test there is no need for the comparison operator.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/flow/FlowGenericProblem_impl.hpp (hunk 2 above)
Comment (GitPaean) [reply opm/simulators/flow/FlowGenericProblem_impl.hpp:110]: Thanks for the information.  It is a good idea to suggest a test for MixingRateControls, but I am not familiar with implementation of MixingRateControls.

Comment (akva2) [reply opm/simulators/flow/FlowGenericProblem_impl.hpp:110]: see https://github.com/OPM/opm-simulators/pull/5543

Comment (GitPaean) [reply opm/simulators/flow/FlowGenericProblem_impl.hpp:110]: Thanks. I will wait for that PR goes in.

**Comment by GitPaean** [comment]
jenkins build this please

**Review by totto82 — APPROVED** [review]
Looks good to me.

**Comment by GitPaean** [comment]
> Looks good to me.

Thanks for approval.  I am getting it in before the jenkins checking gets outdated.

**Comment by GitPaean** [comment]
It is a relatively straightforward change. If it happens later that we want to do differently when coming to design, it is not difficult to change it in different way.  But I think it should have been in FlowProblem in the first place due to its implementation dependence on blackoil setup.

**Pull request merged**